compaction/compaction_manager.cc

```diff
@@ -550,6 +550,14 @@ sstables::shared_sstable compaction_manager::sstables_task::consume_sstable() {
     return sst;
 }
 
+future<semaphore_units<named_semaphore_exception_factory>> compaction_manager::task::acquire_semaphore(named_semaphore& sem, size_t units) {
+    return seastar::get_units(sem, units, _compaction_data.abort).handle_exception_type([this] (const semaphore_aborted& e) {
```

Comment:
I'm not proposing to change it here, but shouldn't `seastar::abort_source:;request_abort()` get the `void requrest_abort(std::exception_ptr)` overload? Or be like semaphore that inherits from custom exception factory and can generate aborted exceptions whatever user wants?

Reply:
it is a nice idea but I don't think it will help here since the abort_source is that of the stop_signal so all the requester knows is that scylla is stopped.
The handler for the abort is the semaphore which is also out of the compaction layer.
Maybe compaction_manager could have a custom semaphore that would generate the compaction_stopped_exception via the semaphore exception factory, though I'm not sure this worth the complexity.

Reply:
@xemul ping

Reply:
> the abort_source is that of the stop_signal so all the requester knows is that scylla is stopped.

Erm... Why is that? The stop-signal abort source is passed into compaction manager by reference and the compaction manager just _subscribes_ on it, so once the stop-signal is fired, compaction manager is called the `do_stop()` and aborts its semaphores with some other abort source (@bhalevy )

Reply:
You're right.